# HG changeset patch
# User Rik <rik@octave.org>
# Date 1374777143 25200
#      Thu Jul 25 11:32:23 2013 -0700
# Node ID cb8543396024485d5b9af2a8126360733d0768dd
# Parent  86067af51d5eaf6da1e304bf6e68a701f6342947
Set minorticks for error bar plots with logarithmic axes.

* scripts/plot/loglogerr.m, scripts/plot/semilogxerr.m,
scripts/plot/semilogyerr.m: Set axis minortick value to "on".

diff --git a/scripts/plot/loglogerr.m b/scripts/plot/loglogerr.m
--- a/scripts/plot/loglogerr.m
+++ b/scripts/plot/loglogerr.m
@@ -45,17 +45,21 @@ function h = loglogerr (varargin)
 
   [hax, varargin] = __plt_get_axis_arg__ ("loglogerr", varargin{:});
 
   oldfig = ifelse (isempty (hax), [], get (0, "currentfigure"));
   unwind_protect
     hax = newplot (hax);
 
     set (hax, "xscale", "log", "yscale", "log");
+    if (! ishold (hax))
+      set (hax, "xminortick", "on", "yminortick", "on");
+    endif
     htmp = __errcomm__ ("loglogerr", hax, varargin{:});
+
   unwind_protect_cleanup
     if (! isempty (oldfig))
       set (0, "currentfigure", oldfig);
     endif
   end_unwind_protect
 
   if (nargout > 0)
     h = htmp;
diff --git a/scripts/plot/semilogxerr.m b/scripts/plot/semilogxerr.m
--- a/scripts/plot/semilogxerr.m
+++ b/scripts/plot/semilogxerr.m
@@ -45,17 +45,21 @@ function h = semilogxerr (varargin)
 
   [hax, varargin] = __plt_get_axis_arg__ ("semilogxerr", varargin{:});
 
   oldfig = ifelse (isempty (hax), [], get (0, "currentfigure"));
   unwind_protect
     hax = newplot (hax);
 
     set (hax, "xscale", "log");
+    if (! ishold (hax))
+      set (hax, "xminortick", "on");
+    endif
     htmp = __errcomm__ ("semilogxerr", hax, varargin{:});
+
   unwind_protect_cleanup
     if (! isempty (oldfig))
       set (0, "currentfigure", oldfig);
     endif
   end_unwind_protect
 
   if (nargout > 0)
     h = htmp;
diff --git a/scripts/plot/semilogyerr.m b/scripts/plot/semilogyerr.m
--- a/scripts/plot/semilogyerr.m
+++ b/scripts/plot/semilogyerr.m
@@ -45,17 +45,21 @@ function h = semilogyerr (varargin)
 
   [hax, varargin] = __plt_get_axis_arg__ ("semilogyerr", varargin{:});
 
   oldfig = ifelse (isempty (hax), [], get (0, "currentfigure"));
   unwind_protect
     hax = newplot (hax);
 
     set (hax, "yscale", "log");
+    if (! ishold (hax))
+      set (hax, "yminortick", "on");
+    endif
     htmp = __errcomm__ ("semilogyerr", hax, varargin{:});
+
   unwind_protect_cleanup
     if (! isempty (oldfig))
       set (0, "currentfigure", oldfig);
     endif
   end_unwind_protect
 
   if (nargout > 0)
     h = htmp;
